ladybug_geometry/geometry3d/face.py

```diff
@@ -671,14 +679,14 @@ def remove_colinear_vertices(self, tolerance):
                   for i, hole in enumerate(self._holes))
         return _new_face
 
-    def flip(self, preserve_clockwise=False):
+    def flip(self, preserve_orientation=False):
```

Comment:
If I understand correctly from `flip` and `reverse` method we are making it possible to create faces that doesn't follow the right hand rule. Is it really needed? I'm currently using the normal direction of face in honeybee-core assuming it always follows the right-hand rule and use that to assign the type of not provided. I know realized that one can create a geometry where normal direction doesn't match this rule. This can cause issues in ceiling / floor detection. Does it something that we can remove and standardize it to match Radiance.

Maybe I should ask why did you put this option available in the first place?

Reply:
On the contrary, `flip` and `reverse` are what allow us to correct for the cases that don't follow the right hand rule by default and there are many ways that you can end up with a Face3D that doesn't follow the right hand rule.  The most obvious is that you just input vertices into `Face3D.__init__` that are clockwise within the input Plane.  However, the [auto-calculated plane that gets created when you only put in vertices](https://github.com/chriswmackey/ladybug-geometry/blob/master/ladybug_geometry/geometry3d/face.py#L1571-L1583) can result in clockwise vertices even when your vertices are actually counterclockwise within the plane you want.  This can happen if the vertices form a concave shape and the starting vertices sit within an "elbow" of the concave shape (resulting in a plane facing the opposite direction of the original one you want).  So we really need these methods in order to help us correct such cases.  BTW, honeybee legacy does not have good checks for these concave elbow cases, which means that it is probably writing many concave shapes with vertices in the wrong order (I have been meaning to go back and fix this at some point).

In any case, the assumption that you use for Radiance is ok and will be correct as long as the Face3D comes from the `counterclockwise_face()` method.  So, across honeybee, we just have to make sure that we always call the `counterclockwise_face()` method when we accept input for a new face. You can see that the `counterclockwise_face()` method is written pretty succinctly to just return `self` if the vertices are already clockwise.  So there's little extra overhead to just calling this method frequently across the library.